src/misc.ts

```diff
@@ -92,3 +92,87 @@ export const hasProperty = (
   object: RuntimeObject,
   name: string | number | symbol,
 ): boolean => Object.hasOwnProperty.call(object, name);
+
+export type PlainObject = Record<number | string | symbol, unknown>;
+
+/**
+ * Predefined sizes (in Bytes) of specific parts of JSON structure.
+ */
+export enum JsonSize {
+  NULL = 4,
+  COMMA = 1,
+  WRAPPER = 1,
+  TRUE = 4,
+  FALSE = 5,
+  ZERO = 1,
+  QUOTE = 1,
+  COLON = 1,
+  DOT = 1,
+}
+
+/**
+ * Check if the value is plain object.
+ *
+ * @param value - Value to be checked.
+ * @returns Boolean.
+ */
+export function isPlainObject(value: unknown): value is PlainObject {
+  if (typeof value !== 'object' || value === null) {
+    return false;
+  }
+
+  try {
+    let proto = value;
+    while (Object.getPrototypeOf(proto) !== null) {
+      proto = Object.getPrototypeOf(proto);
+    }
+
+    return Object.getPrototypeOf(value) === proto;
+  } catch (_) {
+    return false;
+  }
+}
+
+/**
+ * Check if character or string is ASCII.
+ *
+ * @param value - String or character.
+ * @returns Boolean, true if value is ASCII, false if not.
+ */
+export function isASCII(value: string) {
+  // eslint-disable-next-line no-control-regex,require-unicode-regexp
+  return /^[\x00-\x7F]*$/.test(value);
```

Comment:
We only need `isASCII` per character right? For that we could do something like: `return char.charCodeAt(0) > 127`, right?

Reply:
Since the only use case for now is per character, I changed it to follow your suggestion in the next commit.